asset_dashboard/static/js/planner.js

```diff
@@ -125,6 +163,62 @@ class PortfolioPlanner extends React.Component {
     })
   }
 
+  savePortfolio(e) {
+    e.preventDefault()
+
+    const data = {
+      name: this.state.portfolio.name,
+      user: props.user_id,
+      phases: this.state.portfolio.projects.map((phase, index) => {
+        return {'phase': phase.key, 'sequence': index + 1}
+      })
+    }
+
+    const [url, method] = this.state.portfolio.id
+      ? [`/portfolios/${this.state.portfolio.id}/`, 'PATCH'] // Update
+      : ['/portfolios/', 'POST'] // Create
+
+    fetch(url, {
+      method: method,
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'X-CSRFTOKEN': Cookies.get('csrftoken')
+      },
+      body: JSON.stringify(data)
+    }).then(response => {
+      console.log(response.ok, response.status, response.statusText)
+
+      if (!response.ok) {
+        console.error(response.status, response.statusText)
```

Comment:
`fetch` has a built-in way of catching errors, i re-learned last week. you could [chain the promises together and ultimately resolve to catch an error if it raises](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#uploading_json_data). i think this would be cleaner and more idiomatic es6. what do you think about that change?

Reply:
Thanks, @smcalilly! Done in https://github.com/fpdcc/ccfp-asset-dashboard/pull/98/commits/e7c44550ef3555b2b4075deb5fe79da679213847.